service/settings.go

```diff
@@ -34,16 +34,14 @@ func (s *SettingsService) Update(ctx context.Context, u model.UpdateSettingsInpu
 		return model.Settings{}, err
 	}
 
-	settings, err := s.Get(ctx, authUserID)
-	if err != nil {
-		return model.Settings{}, err
-	}
+	settings, err := s.repository.Update(ctx, func(s model.Settings) (model.Settings, error) {
+		if err := s.Update(u); err != nil {
+			return model.Settings{}, apierrors.NewSettingsUnprocessableError().Wrap(err).WithMessage(err.Error())
+		}
```

Comment:
Reading settings and updating it is done in one transaction.

Reply:
Repository returning new `settings` is fine. The rest also looks OK now.

Reply:
(optional) Regarding one transaction, good ambition, and the current implementation is probably fine, albeit perhaps too low-level.
But do we really need it? What's the worst case and what are the potential consequences?
Or as a compromise, consider using `SELECT ... FOR UPDATE`.

Reply:
Thanks 👍 I've decided to try the approach with transactions to practice it. I'm going to keep the current implementation, but I'll add a comment to consider a implementation you suggested. Once I gain more experience while implementing the repository for this app, I might switch to a simpler approach.